src/graphql/client.ts

```diff
@@ -1,6 +1,23 @@
-import { ApolloClient, InMemoryCache } from '@apollo/client';
+import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
+import { setContext } from '@apollo/client/link/context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const httpLink = createHttpLink({
+  uri: 'https://template-onboarding-node-sjz6wnaoia-uc.a.run.app/graphql',
+});
+
+const authLink = setContext(async (_, { headers }) => {
+  const token = await AsyncStorage.getItem('token');
+
+  return {
+    headers: {
+      ...headers,
+      authorization: token ? `${token}` : '',
```

Comment:
aqui vc não precisar token em uma string se ele já for uma strings certo, então daria só pra validar se ele existe, se ele não existir manda uma string vazia, da pra fazer isso com o nullish coalescing operator (`??`), ele verifica se o valor da esquerda é nullish (null/undefined) ou não, se for nullish ele manda o da direita se não for ele mandar o da esquerda mesmo

```suggestion
      authorization: token ?? '',
```